src/Idris/Elab/Interface.idr

```diff
@@ -246,11 +246,13 @@ getConstraintHint : {vars : _} ->
                     Name -> Name ->
                     (constraints : List Name) ->
                     (allmeths : List Name) ->
-                    (params : List Name) ->
+                    (params : List (Name, RigCount, RawImp)) ->
                     (Name, RawImp) -> Core (Name, List ImpDecl)
 getConstraintHint {vars} fc env vis iname cname constraints meths params (cn, con)
-    = do let ity = apply (IVar fc iname) (map (IVar fc) params)
-         let fty = IPi fc top Explicit Nothing ity con
+    = do let pNames = map fst params
+         let ity = apply (IVar fc iname) (map (IVar fc) pNames)
+         let fty = mkTy fc Implicit (map jname params) $
+                   mkTy fc Explicit [(Nothing, top, ity)] con
          ty_imp <- bindTypeNames fc [] (meths ++ vars) fty
```

Comment:
Do I need to add parameters?
```idris
ty_imp <- bindTypeNames fc [] (pNames ++ meths ++ vars) fty
```
like in `mkIfaceData`
https://github.com/idris-lang/Idris2/blob/457ca7c6effd66a3f49e721171b142f979bb7406/src/Idris/Elab/Interface.idr#L114
and `getMethDecl`
https://github.com/idris-lang/Idris2/blob/457ca7c6effd66a3f49e721171b142f979bb7406/src/Idris/Elab/Interface.idr#L152

Reply:
It seems to me that this might be necessary since I added parameters to `fty`, but I can't think of an example where this would have an impact